### PR TITLE
Feature/add update notififier

### PR DESCRIPTION
**Pull request by kasenvr**
My branch naming is impeccable.

This PR...
* Creates a networkingConstants.js file in the libraries folder for Interface. @Misterblue ;)
* Updates the chat to accept a new notification type: System
* Adds a semantic versioning function to utils.js in the libraries folder.
* Creates a version update checker that relies on all of the above to function, notifying you whenever the script is loaded.
* Also mighta fixed an annoying spammy error for equipEntity.js

image

*NOTE:* Undo commit that hard codes in version number before merging.

### CODE REVIEW

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR658/Vircadia-Alpha-PR658-a85391b_abac59c0.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR658/Vircadia-Alpha-PR658-a85391b.tar_abac59c0.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR658/Vircadia-Alpha-PR658-a85391b_abac59c0.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR658/Vircadia-Alpha-PR658-a85391b.tar_abac59c0.gz

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR658/Vircadia-Alpha-PR658-a85391b_abac59c0.exe

File: scripts/system/checkForUpdates.js
Comment (ctrlaltdavid) [inline scripts/system/checkForUpdates.js:14]: Rather than hard-coding this in - requiring an update each release - how about using the values from the `HifiAbout` API?

File: scripts/system/checkForUpdates.js
Comment (ctrlaltdavid) [inline scripts/system/checkForUpdates.js:34]: A better name for `retrievedMetadata` might be `vircadiaMetadata`.

File: scripts/system/checkForUpdates.js
Comment (ctrlaltdavid) [inline scripts/system/checkForUpdates.js:37]: This doesn't check for "dev" version if `currentVersion` is hard-coded, unless `currentVersion` is changed to `"dev"` on `master` and only set to actual version number on release branch. Which would be awkward.

File: scripts/system/libraries/networkingConstants.js
Comment (ctrlaltdavid) [inline scripts/system/libraries/networkingConstants.js:10]: Is the metadata related only to Interface, or does it also relate to the Sandbox server?

File: scripts/system/libraries/utils.js
Comment (ctrlaltdavid) [inline scripts/system/libraries/utils.js:544]:
```suggestion
                if (splitA < splitB) {
                    return -1;
                } else if (splitA > splitB) {
                    return 1;
                }
```
The use of `continue` statement seems awkward.

File: scripts/system/libraries/utils.js
Comment (ctrlaltdavid) [inline scripts/system/libraries/utils.js:553]:
```suggestion
            if (versionAParts[i] < versionBParts[i]) {
                return -1;
            } else if (versionAParts[i] > versionBParts[i]) {
                return 1;
            }
```

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

File: scripts/system/checkForUpdates.js
Comment (kasenvr) [reply scripts/system/checkForUpdates.js:14]: Yup! That's commented out precisely for testing, the commit says to "revert" it once we're ready to merge. :)

File: scripts/system/libraries/networkingConstants.js
Comment (kasenvr) [reply scripts/system/libraries/networkingConstants.js:10]: I've only ever considered it related to the interface but technically the domain-server gets bumped when the interface does so we can in fact update the domain-server to respond to this as well.

File: scripts/system/checkForUpdates.js
Comment (kasenvr) [reply scripts/system/checkForUpdates.js:37]: It will use HifiAbout once live, I hardcoded the value to show the effect for now. :)

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR658/Vircadia-Alpha-PR658-0e8f116_1e15458c.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR658/Vircadia-Alpha-PR658-0e8f116.tar_1e15458c.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR658/Vircadia-Alpha-PR658-0e8f116.tar_1e15458c.Z

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR658/Vircadia-Alpha-PR658-0e8f116_1e15458c.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR658/Vircadia-Alpha-PR658-0e8f116_1e15458c.exe

File: scripts/system/libraries/networkingConstants.js
Comment (ctrlaltdavid) [reply scripts/system/libraries/networkingConstants.js:10]: Add an issue as a reminder to do this?

**Review by ctrlaltdavid — APPROVED** [review]
"Do not merge" label added as reminder to undo commit that hard codes in version number.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR658/Vircadia-Alpha-PR658-37ea33c_ee554ad8.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR658/Vircadia-Alpha-PR658-37ea33c.tar_ee554ad8.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR658/Vircadia-Alpha-PR658-37ea33c.tar_ee554ad8.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR658/Vircadia-Alpha-PR658-37ea33c_ee554ad8.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR658/Vircadia-Alpha-PR658-37ea33c_ee554ad8.exe

**Pull request merged**
